Typos & basic grammar errors fixing

Fourchaux opened the pull request:
This PR is a "rewriting" of ocaml#817

To resume:
Inside comments of ml {i} files, only indisputable (uhuh...) typos
as well as some basic grammar errors are corrected.

Directories checked :
asmcomp - bytecomp - debugger - lex - ocamldoc - parsing - stdlib - typing - utils.

Certainly, typos are still present.
Please report.

Octachron committed:
ocamldoc: avoid nested <pre> in module description

mmottl committed:
Synchronized Pervasives implementation between stdlib and threads

dra27 committed:
Allow backslashes in FLEXLINK_FLAGS

Also in MKEXE, MKDLL and MKMAINDDLL as a result.

dra27 committed:
Install bootstrapped FlexDLL objects to flexdll/

The install-flexdll target now puts the object files for FlexDLL in a
subdirectory flexdll of the Standard Library instead of in the Standard
Library itself.

A configuration tweak means that -I +flexdll is effectively added to all
compiler invocations and also a pseudo-option -L+flexdll to ocamlmklib
calls to Config.mkdll which fixes PR#7373.

dra27 committed:
Split FlexDLL bootstrap into flexdll and flexlink

Allows the C object files to be provided separately and just the
flexlink binary itself to be bootstrapped.

dra27 committed:
Fix tests/warnings/w04.ml on Windows

dra27 committed:
Ensure testsuite works with OCAMLRUNPARAM=b

Ensure that tests which analyse exception messages explicitly disable
backtrace printing.

dra27 committed:
Ensure testsuite works with OCAMLRUNPARAM=b

Ensure that tests which analyse exception messages explicitly disable
backtrace printing.

Octachron committed:
PR#7478: ocamldoc, avoid module preambule repetion

To avoid module preamble repetition, ocamldoc only use as a module
preamble documentation comments that occur before any module elements,
which should also prevent some unexpected module preamble when the first
documentation comments appears in the middle of the source file.

Octachron committed:
Factorize first comment before elements code

Octachron committed:
Update GPR#1037 changes

dra27 committed:
MPR#7385: fix Unix.stat timestamps during DST

The Win32 API used treats DST as a timezone offset rather than a bias,
resulting in all timestamps being an hour out when the system date is
during DST.

Octachron committed:
PR#7331: updated Makefile

stedolan committed:
Fix optimisation of switches to lookup tables (#1069)

Octachron committed:
PR#7331: update changes

dbuenzli committed:
MPR#7500: remove Uchar.dump (#1081)

and make the Uchar independent from the Format and Printf
modules. Previously this made it impossible to use the type in the
natural habitat that the String, Bytes and Buffer modules could be.

stedolan committed:
Fix Gc.minor_{words,free} by removing [@@noalloc]

dra27 committed:
Move INT64_LITERAL from byterun/ints.c to m.h

gasche committed:
Merge pull request #1085 from Octachron/ocamldoc_self_alias_loop

PR#7331: ocamldoc, avoid self-alias induced infinite loop

ygrek committed:
Changes: fix GPR# typo

gasche committed:
[minor] 4.05 Changes cleanup

gasche committed:
another round of 4.05 Changes fixes

dra27 committed:
Don't build flexlink.opt in opt.opt

Require flexlink.opt to be specified. Old behaviour:

make flexdll world.opt install

now becomes:

make flexdll world.opt flexlink.opt install

Without flexlink.opt, a bytecode version of flexlink.exe is installed.

dra27 committed:
Fix installation of bytecode flexlink.exe

The bytecode version of flexlink.exe is compiled before camlheader is
available, so it needs to have that prepended when it's installed.

objmagic committed:
Support inherited field in object type expression.

Also add location on object field label and polymorphic variant tags.

mshinwell committed:
Allow creation of empty .cmxa files on macOS

hannesm committed:
Respect OCAML_COLOR environment variable for deciding whether to use colors

Since 4.03, OCaml supports coloring its messages to standard output and standard
error, depending on the "-color" argument ({always,never,auto}).  This commit
adds support for the environment variable "OCAML_COLOR" (which value can as well
be {always,never,auto}).

The command line argument "-color" takes precedence, "OCAML_COLOR" is only
taken into consideration if no "-color" is provided.

The motivation for this is that the user should have control over coloring
OCaml's output messages.  OCamlbuild, a widely used build tool executes OCaml
not under a tty (and OCaml does not colorize errors and warnings), which lead
various packages use `color(always)` in their `_tags` files, which breaks with
other (non-interactive) programs (i.e.  editor helpers).

Further discussion was done at https://github.com/ocaml/ocamlbuild/issues/87 and
https://github.com/ocaml/ocaml/pull/1098.

gasche committed:
Revert "Deprecate Bigarray.*.map_file and add Unix.map_file (#997)"

This reverts commit 5ed72007f8185b2bd3c7a663abdcb1be5eaeee7c.

GPR#997 introduced a hard dependency of Bigarray on Unix, while there
previously only was a type-level dependency. This break some programs,
such as Camomile, that linked bigarray.cma but not unix.cma.

A solution is being worked out in GPR #1077 to remove the dependency,
but I would like to go forward with opam package testing on the 4.05
branch, and getting the details right for GPR#1077 requires some care,
so I wouldn't feel comfortable rushing to merge it.

I had to handle the following conflicts:
	otherlibs/unix/unix.mli
          ("@since 4.05.0" was added in faab91a96c1195aac925d75f9127d0a698230e70)
	testsuite/tests/lib-bigarray-file/mapfile.ml
          (changed by 5839c9827d9fdd55e5a9eff6bbd6173370c30bbb;
           I kept and adapted the new version)
	testsuite/tests/lib-bigarray-file/mapfile.reference

xavierleroy committed:
Merge pull request #1102 from gasche/revert-unix-bigarray

Revert "Deprecate Bigarray.*.map_file and add Unix.map_file (#997)"

This revert is intended to unblock the OPAM package testing of 4.05.

garrigue committed:
add test for PR#6987

xavierleroy committed:
Fix Unix.create_process in the case where the same descriptor is used several times

As reported in GPR#1105, the current implementation (from commit ab4e3be) fails if the same descriptor is passed several times to Unix.create_process, e.g. as the new stdout and the new stderr (a la 2>&1 shell redirection).

This is a tentative, simple fix, lighter than the 4.04 implementation and the GPR#1105 proposed implementation.  A test was added.

xavierleroy committed:
Fix Unix.create_process in the case where the same descriptor is used several times (continued)

The lightweight fix in commit c32773d wasn't good enough if the repeated descriptor is 0, 1 or 2, e.g. Unix.create_process ...  Unix.stdin Unix.stdout Unix.stdout.

By lack of better ideas, this commit reverts to the 4.04 implementation of perform_redirections.

The test suite was updated with the Unix.create_process ...  Unix.stdin Unix.stdout Unix.stdout case above.

Octachron committed:
PR#7504: fix warning 8 with unconstrained records

gasche committed:
Merge pull request #1106 from Octachron/pr7504_fix

PR#7504: fix warning 8 with unconstrained records

mshinwell committed:
Fix hash table performance problem in Emitaux (PR#7456) (#1092)

Octachron committed:
PR#7497: manual, numbered tables of contents

Octachron committed:
Changes for GPR#1095

dra27 committed:
Fix \r problems on recent Cygwins

The Cygwin packages for awk, grep and sed were updated on 20 February
2017 so that they no longer automatically strip \r characters on binary
mounts. This affects the OCaml build system in a few places, requiring
the addition of a few tr calls.

References:
  * https://cygwin.com/ml/cygwin/2017-02/msg00152.html
  * https://cygwin.com/ml/cygwin/2017-02/msg00189.html
  * https://cygwin.com/ml/cygwin/2017-02/msg00188.html

xavierleroy committed:
Fix Unix.create_process in the case where the same descriptor is used several times (continued)

After the revert to 4.04 implementation in commit d54f631, Damien Doligez came up with a very nice solution, see
https://github.com/ocaml/ocaml/pull/1105#issuecomment-286455646

This commit adopts Damien's solution, and adds a test for swapping stdout and stderr.

damiendoligez committed:
last commit before tagging 4.05.0+beta3

damiendoligez committed:
change VERSION for 4.05.0+beta3

damiendoligez committed:
increment version number after tagging 4.05.0+beta3

dra27 committed:
Don't emit \r in the debugger

dra27 committed:
Turn C warnings into errors on AppVeyor

dra27 committed:
Fix missing types in win32graph/draw.c

xavierleroy committed:
Remove dependencies from objinfo_helper to OCaml runtime (continued)

Removal in commit 2c3a9d3 was too brutal, "config/s.h" must be included so that HAS_BFD is seen.

gasche committed:
Changes entry for Sébastien's Makefile factorization work

gasche committed:
Changes: document merge-makefile-related changes to Makefile.config

dra27 committed:
Fix missing header on MSVC

dra27 committed:
Switch AppVeyor to Visual Studio 2015

Visual Studio 2010 x64 is supposed to enable /Wp64 (detect 64-bit
porting issues) by default but this does not work. Adding /Wp64 will add
a lot of deprecation warnings to the log; but not having /Wp64 has
caused a bug in otherlibs/win32unix/channels.c to go unnoticed.
Therefore, switch to Visual Studio 2015 to workaround the problem.

dra27 committed:
Update docs for Visual Studio 2017

dra27 committed:
Build the FlexDLL objects manually on AppVeyor

VS2015 isn't supported by the binary release of FlexDLL.

objmagic committed:
Merge branch 'trunk' into object-inherit

objmagic committed:
Add another test.

objmagic committed:
Remove message.

objmagic committed:
Add Change entry.

objmagic committed:
Fix bug when handling empty object type exp

Also add tests.

objmagic committed:
Update Changes entry.

objmagic committed:
Unify, rather than copy.

dra27 committed:
Test msvc32 on AppVeyor using Visual Studio 2010

murmour committed:
Fixed erroneous comments

murmour committed:
Fixed grammar in user-facing comments and README

murmour committed:
Fixed erroneous comments

murmour committed:
Fixed grammar in error messages

objmagic committed:
Correct error message, more tests.

objmagic committed:
Update ocamldoc tests

Also add test for polymorphic variants.

objmagic committed:
Update Changes entry.

objmagic committed:
Really correct the error message.

objmagic committed:
Missing arrow.

Octachron committed:
Manual: fix link to Ast mapper module

gasche committed:
Merge pull request #1129 from Octachron/manual_missing_link

Manual: fix link to Ast mapper module

damiendoligez committed:
PR#7505: Memory cannot be released after calling Bigarray.Genarray.change_layout.

damiendoligez committed:
update Changes for PR#7505

dra27 committed:
Wallpaper over \r\n problem for Cygwin testsuite

This allows the Cygwin testsuite to pass on a Windows (CRLF) checkout of
the OCaml sources.

robblanco committed:
Fix typo in coreexamples.etex

shindere committed:
Add a new simpler CI build script

This script is used on Inria's infrastructure

Once all Inria CI jobs will use this script, it will replace the
legacy one.

(cherry picked from commit d847e077e8d9f5f434e630c05977b8657de5a57d)

gasche committed:
Changes fixes

dra27 committed:
Switch AppVeyor to use Cygwin64

shindere committed:
In the new CI script, enable build of runtime with debugging support

This is done by passing the -with-debug-runtime option to the configure
script on the non-Windows systems.

On Windows, a sed invocation is used to set RUNTIMED to true in
config/Makefile.

(cherry picked from commit fdc63806c3c243267e3e82207971794bdd4128ca)

shindere committed:
On Linux, also build the instrumented runtime during CI

(cherry picked from commit 878508c6ba3d2d25ec25c5f646b73af1275abd10)

shindere committed:
Add support for building and testing with flambda to the new CI script

If the flambda environment variable is set to true, the feature is
enabled in the build system.

(cherry picked from commit 24b028c1fc433e46772ec19ec7adc5e2cc4d0100)

shindere committed:
Replace legacy CI build script by new one

Since there are no jobs left on Inria's CI that use the legacy
ci-build script, replace it by the new one.

(cherry picked from commit 7d6a6614712d1e25f27be989d83d5bc1410c2c4a)

shindere committed:
Fix Windows installation directory in tools/ci-build

This commit sets the installation directory for the Windows build to the
same value as in the config/Makefile.m* files.

So, in itself this commit is a no-op. It however introduces the next one
which will ensure ci-build's instdir setting is honored even for
the builds that do not use configure.

shindere committed:
tools/ci-build: honor instdir on non-configure builds

The instdir environment variable is used by ci-build to define where
make install should install the compiler. Before this commit, a value was
assigned to instdir for the Windows builds that do not use configure,
but then this value was not taken into account, making the assignments useless
for those builds that do not use configure.

This commit fixes this by using instdir to override the value of PREFIX
in config/Makefile.

shindere committed:
tools/ci-build: install to a different directory for each build

This commit ensures the install directory will be distinct for each build,
with the two following motivations:

1. If required, it will make it possible to have several builds running
concurrently on the same slave without conflicts between their make install

2. So far, if for any reason make install fails, then the build script
gets stopped and the install directory will not be removed.
Thus if a future build tries to install to the same directory, its
installation could in principle fail, not because it is broken but because
the previous build has left the install directory in a corrupted state.

So this commit ensures that make install will be done in an empty
directory.

shindere committed:
Fix tools/ci-build

No spaces allowed in varialbe assignments.

damiendoligez committed:
small typo in manual

mshinwell committed:
PR#7533

mshinwell committed:
Changes

gasche committed:
Merge pull request #1173 from mshinwell/pr7533-4.05

Correctly perform side effects for "/" and "mod" (PR#7533)

gasche committed:
[minor] Changes cleanup

dra27 committed:
Don't remove generated .s in tests/asmcomp

dra27 committed:
Suppress cl output of source file

dra27 committed:
Missing @ prefixes in testsuite

dra27 committed:
Fix cloexec test on 64-bit Windows

This test was passing because handles on 64-bit Windows tend to use the
low 32-bits only... but there's no guarantee!

objmagic committed:
Merge branch 'trunk' into object-inherit

Octachron committed:
Manual: fix dead links in ocamldoc chapter

gasche committed:
Merge pull request #1181 from Octachron/manual_ocamldoc_dead_links

MPR#7539: manual, fix dead links in the ocamldoc chapter

mshinwell committed:
Functors

gasche committed:
Merge pull request #1177 from dra27/testsuite-tweaks

Testsuite tweaks

mshinwell committed:
Add missing testcase

damiendoligez committed:
Cherry-pick 27fccad from trunk:

GPR#1150: Fix typo in arm64 caml_alloc3 assembler directives

yallop committed:
Int(32|64).of_string: accurately document the exception raised.

yallop committed:
Int(32|64).of_string and int_of_string: document the 0u prefix.

garrigue committed:
Fix PR#7348

dra27 committed:
Display CPU name in AppVeyor builds

Skylake/Kaby Lake microcode bug investigation

dra27 committed:
Display backtraces in AppVeyor builds

mshinwell committed:
Erroneous code transformation at partial applications (MPR#7531) (#1162)

Fourchaux committed:
Typos & basic grammar errors fixing

Fourchaux committed:
Typos & basic grammar errors fixing

gasche committed:
manual: build using ocamldoc.opt when available

On my machine, "make clean; make html" goes from 30s to 11s

gasche committed:
manual: fix parallel make in manual/library

The rule said:

$(INTF): $(MLIS)
	$(OCAMLDOC) -latex [...] $(MLIS)
        # renaming of files
       	mv -f Arith_status.tex Arithstatus.tex
	mv -f Big_int.tex Bigint.tex
	mv -f Ast_helper.tex Asthelper.tex
	mv -f Ast_mapper.tex Astmapper.tex

If two targets executed at the same time depend on different files in
INTF, this rule would be executed several time in parallel, and in
particular the calls to `mv -f ...` would race with each other,
resulting in a build failure.

Now the $(OCAMLDOC) call is made unique by using a single target:

$(INTF): interfaces
interfaces: $(MLIS)
	$(OCAMLDOC) -latex [...] $(MLIS)
        [...]

Note that parallel build does not improve the build speed on my
machine; but it can still make people more productive by not breaking.

gasche commented:
The typos fixed seem reasonable to me, thanks!

Do we know why the testsuite fails on Windows? (Is this a line-ending issue?)

It would be nice of you to add a Changes entry in either the "Manual and documentation" or the "Internal/compiler-libs changes", crediting your work and also the people that provided reviews and feedback.

Fourchaux committed:
Add a Changes entry

Fourchaux committed:
Add a Changes entry

gasche committed:
GPR#1203: Changes entry

gasche committed:
Merge pull request #1203 from gasche/manual-build-perf

Make the manual build faster

dra27 commented:
AppVeyor fail is my fault, and unrelated (test should be restarting shortly).

This all looks good to me, thanks - it's ready for @damiendoligez to merge after 4.05.0 is released.

mshinwell committed:
Call counts in Spacetime (#1180)

Octachron committed:
GPR#1205: manual, remove last remnants of labltk (#1205)

yallop committed:
int_of_string: document the behaviour of 0u with high unsigned values.

gasche committed:
Merge pull request #1198 from yallop/of_string-documentation

Int(32|64).of_string documentation: exceptions and prefixes

Octachron committed:
Documentation: fix references in format and scanf

The format and scanf modules were sometimes using `[!ref]` or
`[{!ref}]` in place of `{!ref}`. Moreover the format modules was
sometimes trying to use reference inside code block, which may be an
interesting feature request for ocamldoc/odoc, but is not currently
implemented in ocamldoc.

Octachron committed:
doc: non-inline code block in format

gasche committed:
Merge pull request #1206 from Octachron/doc_fix_references

Standard library documentation: fix few references

murmour commented on typing/ctype.ml:
```diff
@@ -1315,7 +1315,7 @@ let subst env level priv abbrev ty params args body =
 
 (*
-   Only the shape of the type matters, not whether is is generic or
+   Only the shape of the type matters, not whether is generic or
```
... not whether _it's_ generic or not.

Fourchaux replied:
Fixed. TY.

murmour commented on stdlib/camlinternalFormat.ml:
```diff
@@ -2055,15 +2055,15 @@ let fmt_ebb_of_string ?legacy_behavior str =
       "non-zero widths are unsupported for %c conversions"
   in
-  (* Raise Failure with a friendly error message about an option dependencie
+  (* Raise a Failure with a friendly error message about an option dependency
```
I wouldn't use an article before "Failure".

damiendoligez replied:
Me neither, and I would put `Failure` in brackets (ocamldoc-style).

Fourchaux replied:
OK. @dra27 also said :
Either Raise [Failure] or "Raise a Failure exception".

Raise [Failure] is the winner :)

Thank you.

murmour reviewed (commented):
Looks good. I've added some minor remarks.

Fourchaux committed:
small typo

Fourchaux committed:
small typo

Fourchaux committed:
Fix Raise [Failure] comment

Fourchaux committed:
Fix Raise [Failure] comment

damiendoligez committed:
PR#7511: Unboxed type with unboxed argument should not be accepted (#1133)

damiendoligez committed:
PR#7511: Unboxed type with unboxed argument should not be accepted (#1133)

yallop committed:
[Tests] print GetLastError() as unsigned long.

yallop commented on ocamldoc/odoc_print.ml:
```diff
@@ -47,7 +47,7 @@ exception Use_code of string
    from the signatures. Used when we don't want to print a too long module type.
    @param code when the code is given, we raise the [Use_code] exception is we
-   encouter a signature, to that the calling function can use the code rather
+   encounter a signature, to that the calling function can use the code rather
```
"is we encouter" ↝ "if we encouter"

"to that" ↝ "so that"

Fourchaux replied:
Fixed. Thank you.

gasche committed:
Merge pull request #1208 from yallop/dword-format

Fix a warning in the tests (use "%lu" for GetLastError())

Fourchaux committed:
typos odoc_print.ml

Fourchaux committed:
typos odoc_print.ml

gasche committed:
tentative fix for PR#7563 (module/extension name clash)

There was a bug when a constructor of the same name as a module alias
was referenced from another module.

gasche committed:
PR#7563: testsuite

gasche committed:
PR#7563: deprecate transl_path and use transl_{module,value,extension,class}_path instead

excellent suggestion by Jacques Garrigue and Leo White

damiendoligez committed:
cherry-pick the fix for MPR#7557 from 4.04

damiendoligez committed:
cherry-pick the fix for MPR#7557 from 4.04

damiendoligez committed:
Add Unix.unsafe_getenv. (See MPR#7557)

gasche committed:
Changes entry

gasche committed:
Merge pull request #1210 from gasche/PR#7563-module-constructor-name-clash

Pr#7563 module constructor name clash

gasche committed:
[minor] harden config/Makefile against '#' in PREFIX

The opam-compiler-conf script will generate an opam switch name
(and thus a directory name) from the name of the current git
branch. Branches named 'PR#1234-foo-bar' would have the ./configure
script generate a config/Makefile with the lines

    PREFIX=~/.opam/4.06.0+local-git-PR#1234-foo-bar
    BINDIR=$(PREFIX)/bin
    BYTERUN=$(BINDIR)/ocamlrun
    LIBDIR=$(PREFIX)/lib/ocaml

The '#' in the first line parses as the start of a comment, so
the second part is ignored and the build system would then install
in ~/.opam/4.06.0+local-git-PR instead.

After this change, config/Makefile starts with:

    # generated by ./configure --prefix ~/.opam/4.06.0+local-git-PR#1234-foo-bar
    CONFIGURE_ARGS=--prefix ~/.opam/4.06.0+local-git-PR\#1234-foo-bar
    PREFIX=~/.opam/4.06.0+local-git-PR\#1234-foo-bar

damiendoligez committed:
Some tweaks for MPR#7557 (#1213)

* fall back to __secure_getenv when secure_getenv is not available

* use secure_getenv for instrumented runtimes

* documentation: warn against setting the setuid or setgid bits on custom bytecode executables

damiendoligez committed:
Some tweaks for MPR#7557 (#1213)

* fall back to __secure_getenv when secure_getenv is not available

* use secure_getenv for instrumented runtimes

* documentation: warn against setting the setuid or setgid bits on custom bytecode executables

damiendoligez committed:
last commit before tagging 4.05.0+rc1

damiendoligez committed:
change VERSION for 4.05.0+rc1

damiendoligez committed:
increment version number after tagging 4.05.0+rc1

yallop committed:
Restrict Unix.environment in privileged contexts; add Unix.unsafe_environment (#1217)

* Restrict Unix.environment to return an empty array in privileged environments.

* Add Unix.unsafe_environment.

Unix.unsafe_environment is an analogue of Unix.environment that
returns the process environment regardless of privileges.

dbuenzli committed:
Clflags: switch default value of keep_locs from false to true.

This provides better error messages from the toolchain by default.

dbuenzli committed:
Bootstrap.

emillon committed:
Fix "-keep-docs" option in ocamlopt manpage

It was a duplicate of `-keep-locs` below.

gasche committed:
Merge pull request #1220 from emillon/typo-manpage-keep-docs

Fix "-keep-docs" option in ocamlopt manpage

gasche committed:
Merge pull request #1219 from dbuenzli/keep-locs-default

Compiler drivers: make -keep-locs the default

yallop committed:
Use a more robust search for no-change-needed lines based on merge-base.

(No change needed.)

vramana committed:
Cleanup HACKING.doc

Makefile.shared doesn't exist anymore.

gasche committed:
Merge pull request #1224 from vramana/patch-1

HACKING.adoc: remove mention of now-gone `Makefile.shared`

yallop committed:
.travis-ci.sh: switch string manipulation from sed to bash.

yallop committed:
Remove the final use of TRAVIS_COMMIT_RANGE.

yallop committed:
Illustrate the relationship between the $TRAVIS_ variables

gasche committed:
Merge pull request #1222 from yallop/dotdotdot

Travis tests: use a more robust search for no-change-needed lines based on merge-base.

dra27 committed:
Clarify Changes entry for GPR#875

dra27 committed:
Clarify Changes entry for GPR#875

Octachron committed:
MPR#248: remember weak variable names

ygrek committed:
get rid of compare_val in String.concat

ygrek committed:
get rid of compare_val in String.concat

dbuenzli committed:
Add Unicode character escape \u{X+} to OCaml string literals.

The syntax of string literals is changed to add the \u{X+} escape
sequence which replaces the escape by the UTF-8 encoding of the
Unicode scalar value denoted by the hexadecimal number H made of one
to six lower or uppercase hexadecimal digits.

Unfortunately due to OCaml lax policy about escape sequences this
change can break program that trigger warning 14 on compilation. More
precisely programs which have literals with "\u" as a substring will
now either fail or the string literal will be silently compiled to
another representation in case the literal has a subsequence that
respects the \u{X+} syntax.

gasche committed:
[minor] update .mailmap

gasche committed:
Merge pull request #1233 from gasche/mailmap-update

[minor] update .mailmap

dra27 committed:
Minor fix to Cygwin testsuite on CRLF-checkouts

gasche committed:
Merge pull request #1234 from dra27/minor-fix

Trivial testsuite fix on Cygwin for CRLF checkouts

trefis committed:
Mention deprecated warning 25 in manual and man page

* warning 25: updated the man page

* warning 25: update the manual

dbuenzli committed:
Lexer escapes: factorize hex number parsing.

Octachron committed:
documentation: margin and max_indent coupling

ngzhian committed:
Clarify ocamldoc -keep-code behavior

damiendoligez committed:
PR#7452: tweak GCC options to try to work around the Skylake/Kaby lake bug (#1228)

gasche committed:
Merge pull request #1238 from ngzhian/trunk

Clarify ocamldoc -keep-code behavior

gasche committed:
Changes: synchronize with the 4.05 branch

alainfrisch committed:
Fix pprintast for infix operators starting with #.

damiendoligez committed:
fix up the merge of 4.05 into trunk

gasche committed:
Merge pull request #1225 from Octachron/remember_weak_types

MPR#248: unique names for weak type variables

gasche committed:
[minor] Changes: GPR#1012 is not compatibility-breaking

xavierleroy committed:
Introduce UnixLabels.map_file

This is a follow-up to the introduction of Unix.map_file in GPR#997 and GPR#1077.

xavierleroy committed:
Merge pull request #1236 from Octachron/document_format_coupling

Stdlib documentation, Format module: format margin and maximum indentation are coupled

yallop committed:
Make git commit manipulation more robust for branch builds.

(no change entry needed)

xavierleroy committed:
Merge pull request #1246 from yallop/travis-ci-branch-pushes

.travis-ci.sh: make git commit manipulation more robust for branch builds.

gasche committed:
Changes: consistently use MPR# instead of PR#

gasche committed:
Changes: remove duplicate GPR#795 entry

(spotted by Octachron)

gasche committed:
Merge pull request #1243 from alainfrisch/fix_pprint_ast

Fix pprintast for infix operators starting with #

gasche committed:
Merge pull request #1245 from ocaml/map_file_omission

Introduce UnixLabels.map_file

damiendoligez committed:
take @shindere's remarks into account

damiendoligez committed:
merge 4.05 into trunk

damiendoligez committed:
Merge pull request #1244 from damiendoligez/merge-4.05-into-trunk

Merge 4.05 into trunk

nojb committed:
Unix.symlink: support forward slashes in symlink targets under Windows (#1211)

* Normalize target before calling CreateSymbolicLink

* Add Changes entry

* Run test only on Windows with symlink permissions

alainfrisch committed:
Try to fix MinGW build.

lpw25 committed:
Fix Env.without_cmis (#1223)

* Fix Env.without_cmis

* Add regression test

* Add changes entry

gasche committed:
Merge pull request #1241 from gasche/sync-trunk-Changes-with-4.05-Changes

Changes: synchronize trunk with the 4.05 branch

damiendoligez committed:
ignore and clean up new generated files in tests/typing-multifile

herbelin committed:
Re-assigning events of a pack of modules to the individual modules of the pack. (#1179)

dbuenzli committed:
Lexer string literals: use Buffer module.

Replace the local implementation of a growable string buffer by
a Buffer.t value.

alainfrisch committed:
Support 'let open' in class and class type expressions (#1249)

* Support 'let open' in class and class type expressions.

* Adapt ocamlprof.

* Adapt ocamldoc.

* Add tests.

* Changelog.

* Manual.

alainfrisch committed:
Merge branch 'trunk' into string-unicode-escapes

alainfrisch committed:
Merge pull request #1232 from dbuenzli/string-unicode-escapes

Add Unicode character escape \u{H} to OCaml string literals.

rnd4222 committed:
Disable cplugins feature by default (#1242)

* Disable cplugins feature by default

* Both --with-cplugins and --no-cplugins configure options

* Added Changes entry for GPR#1242

jacquev6 committed:
Reference manual: fix syntax errors in grammar

jacquev6 committed:
Reference manual: re-add \\

As noted by @Octachron

jacquev6 committed:
Reference manual: fix grammar in Language extensions

No change entry needed

Octachron committed:
Merge pull request #1256 from jacquev6/trunk

Reference manual: fix grammar in Language extensions

xavierleroy committed:
MPR#7591: frametable not 8-aligned on x86-64 port

Misalignment was due to the "D.long (const 0)" emitted just before the frametable, which sets the data pointer to 4 mod 8. Looks like someone cut-and-pasted from i386 to amd64 without thinking.

This commit fixes the bug twice (because belt and suspenders and all that) in two obvious ways:
- the data terminator D.long becomes D.qword
- explicit 8-alignment is requested before emitting the frame table.

(Mental note: why is the frame table in the data segment and not in a readonly data segment?)

xavierleroy committed:
MPR#7591, GPR#1257: Update Changelog

objmagic committed:
merge trunk and update test

objmagic committed:
add tests

objmagic committed:
fix new test

gasche committed:
Merge pull request #1257 from ocaml/pr7591

MPR#7591: frametable not 8-aligned on x86-64 port

gasche committed:
cleanup trunk Changes

There seems to be many instances of merges introducing duplicated Changes lines recently,
this may be an unfortunate effect of the use of the 'union' merge driver
since 4d2f22f269fda0aeb7d6c547ef366fd4661008b6.

gasche committed:
Merge pull request #1214 from gasche/escape-Makefile-injections

[minor] harden config/Makefile against '#' in PREFIX

gasche committed:
Merge pull request #1258 from gasche/cleanup-Changes

cleanup trunk Changes

olafhering committed:
MPR#7281: fix .TH macros in generated manpages

The troff .TH macro takes up to three extra args, according to the
groff documentation at gnu.org. ocamldoc inserts a fourth argument
"source:". Remove the extra argument. Fixes commit a87c3f20e
("Enable ocamldoc to build reproducible manpages")

Without this change the header/footer of each ocaml manpage is
broken. Example: the date should be in the middle of footer,
instead of the left side. The string "source:" should not be shown
at all. No other manpage has such string.

Signed-off-by: Olaf Hering <olaf@aepfle.de>

Octachron committed:
Merge pull request #1259 from olafhering/mpr7281

MPR#7281: fix .TH macros in generated manpages

alainfrisch committed:
Describe workflow for contributions (#1260)

damiendoligez committed:
Fix subtle syntax error in test output that makes the scripts ignore a failed test; also fix the failing test

xavierleroy committed:
Merge pull request #1263 from damiendoligez/fix-testsuite-bug

Fix testsuite bugs

gasche committed:
Merge pull request #1118 from objmagic/object-inherit

Support inherited field in object type expression

gasche committed:
parsetree.mli: consistently use 'label' for polymorphic variants and objects

PR #1118 had turned polymorphic variant constructors from 'label' to
its definition 'string' for consistency with field names in object
types; instead, we consistently name 'label' the method and instance
variable names throughout the AST. This does not break compatibility
as the two types are synonym, but it should improve readability of
parsing/parsetree.mli.

gasche committed:
travis: temporarily disable campl4 build broken by GPR#1118

xavierleroy committed:
Merge pull request #1265 from gasche/temporarily-disable-camlp4

travis: temporarily disable campl4 build broken by GPR#1118

Commit:
Stop building camlp4 in Travis

Commit:
Stop building ocamlbuild in Travis

and remove commented out building of opam

xavierleroy committed:
Merge pull request #1270 from diml/bye-bye-camlp4

Permanently stop building Camlp4 in Travis builds

mshinwell committed:
Fixes for out-of-range Ialloc

mshinwell committed:
Merge remote-tracking branch 'ocaml/trunk' into functor_currying-trunk

mshinwell committed:
Reduce size of test case

yallop committed:
Fix some duplication in Changes introduced by bad merging. (#1267)

xavierleroy committed:
Merge pull request #1271 from mshinwell/ialloc

Fixes for Ialloc instructions allocating more than Max_young_wosize words in the minor heap

Out-of-range Ialloc instructions cause various problems, see in particular GPR #1250.

gasche committed:
Merge pull request #1264 from gasche/parsetree-use-label

parsetree.mli: consistently use 'label' for polymorphic variants and objects

chambart committed:
Optimize away some physical equality (#850)

mshinwell committed:
Fixes

mshinwell committed:
Merge remote-tracking branch 'ocaml/trunk' into functor_currying-trunk

mshinwell committed:
Update test

mshinwell committed:
Spacetime configure test

gasche committed:
Merge pull request #1275 from mshinwell/spacetime_configure

Spacetime configure test

lpw25 committed:
Merge pull request #1183 from mshinwell/functor_currying-trunk

Correct handling of inlining attributes for multi-argument functors

chambart committed:
Tweak cmmgen debuginfo in presence of optimisations (#873)

yallop committed:
String_val: return 'const char *' when -safe-string is globally enabled (#1274)

Chris00 committed:
Make clear "toplevel" = "REPL" in the TOC (#1276)

swalk-cavium committed:
Aarch64 - enable large immediate offset (#1250)

kunigami committed:
nit: improve List.rev_append doc wording (#1197)

mshinwell commented:
@Fourchaux please rebase.  It's high time we got this merged.

gasche committed:
Merge pull request #1037 from Octachron/ocamldoc_preamble_duplication

PR#7478: ocamldoc, avoid module preamble repetition

gasche committed:
.gitattributes: disable the 'union' merge driver for Changes (#1277)

chambart committed:
Fix MPR#7259 by implementing switch branch sharing for flambda (#603)

Fourchaux committed:
Fix typedtree.ml/mli

mshinwell commented:
Looks like something's gone wrong with that rebase?

Fourchaux commented:
Not sure what was wrong (incorrect git push I think) !

My 'github.com:Fourchaux/ocaml.git' is good now (I hope ...)


But this PR  displays '...merge ... commits into ocaml:trunk from unknown repository'.
How to solve that? Should I open a new PR?

gasche commented:
Yes, feel free to open a new PR.
